kernel/src/hil/public_key_crypto/keys.rs

```diff
@@ -283,3 +283,98 @@ pub trait RsaPrivKeyMut: PubPrivKeyMut + RsaKeyMut {
     /// the output of this function.
     fn take_exponent(&self) -> Option<&'static mut [u8]>;
 }
+
+/// Client for selecting keys.
+pub trait KeySelectClient {
+    /// Called when the specified key is active and ready to use for the next
+    /// cryptographic operation.
+    ///
+    /// ### `error`:
+    ///
+    /// - `Ok(())`: The key was selected successfully.
+    /// - `Err(())`: The key was selected set successfully.
+    ///   - `ErrorCode::INVAL`: The index was not valid.
+    ///   - `ErrorCode::FAIL`: The key could not be set.
+    fn select_key_done(&self, index: usize, error: Result<(), ErrorCode>);
+}
+
+/// Interface for selecting an active key among the number of available keys.
+///
+/// This interface allows for the implementer to maintain an opaque internal
+/// representation of keys. They may be stored in memory, flash, or in secure
+/// element (where the actual key may not be accessible). Users of this
+/// interface can select which key is active for cryptographic operations. There
+/// is no assumption for implementers of this interface that keys can be added
+/// or changed or that keys can be specified by their actual bytes in a slice.
+///
+/// Selecting a key is asynchronous as it may require communication over a bus
+/// or waiting for an interrupt.
+///
+/// Keys are specified by an index starting at zero and going to
+/// `get_key_count()-1`.
+///
+/// The stored keys can be public or private keys.
+pub trait KeySelect<'a> {
```

Comment:
To be in line with the possible `KeySet` renaming, maybe rename this to `SelectKey`?

Reply:
Renamed.